internal/provider/resource_access_model.go

```diff
@@ -5,186 +5,314 @@ import (
-				Type:        schema.TypeString,
 				Required:    true,
-				ForceNew:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
 			},
-			"users": {
+			"users": schema.ListAttribute{
 				Description: "List of users to grant access to",
-				Type:        schema.TypeList,
 				Required:    true,
-				Elem: &schema.Schema{
-					Type: schema.TypeString,
+				ElementType: types.StringType,
+			},
+			"access": schema.StringAttribute{
+				Description: "Type of access to the model",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					stringvalidator.OneOf("admin", "read", "write"),
 				},
 			},
-			"access": {
-				Description:  "Type of access to the model",
-				ValidateFunc: validation.StringInSlice([]string{"admin", "read", "write"}, false),
-				Type:         schema.TypeString,
-				ForceNew:     true,
-				Required:     true,
+			// ID required by the testing framework
+			"id": schema.StringAttribute{
+				Computed: true,
 			},
 		},
 	}
 }
 
-func resourceAccessModelCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	client := meta.(*juju.Client)
+// Configure enables provider-level data or clients to be set in the
+// provider-defined DataSource type. It is separately executed for each
+// ReadDataSource RPC.
+func (a *accessModelResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
 
-	var diags diag.Diagnostics
+	client, ok := req.ProviderData.(*juju.Client)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *juju.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+	a.client = client
+}
 
-	model := d.Get("model").(string)
-	access := d.Get("access").(string)
-	usersInterface := d.Get("users").([]interface{})
-	users := make([]string, len(usersInterface))
-	for i, v := range usersInterface {
-		users[i] = v.(string)
+func (a *accessModelResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	// Check first if the client is configured
+	if a.client == nil {
+		addClientNotConfiguredError(&resp.Diagnostics, "create")
+		return
 	}
+	var plan accessModelResourceModel
 
-	uuid, err := client.Models.ResolveModelUUID(model)
-	if err != nil {
-		return diag.FromErr(err)
+	// Read Terraform configuration from the request into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
 	}
 
+	// Get the users
+	var users []string
+	plan.Users.ElementsAs(ctx, users, true)
+	// users := convertRawUsers(plan.Users.Elements())
```

Comment:
The error when I first ran was: `Error: target must be a pointer, got []string, which is a slice`

This change is working for me, I've run the tests many times and local plans.
```suggestion
	resp.Diagnostics.Append(plan.Users.ElementsAs(ctx, &users, false)...)
	if resp.Diagnostics.HasError() {
		return
	}
```

Reply:
I'm unclear if we should have `allowUnhandled` as true or false.